### PR TITLE
add Movability to Generator

### LINKED ISSUE #618: Generator followup work
A few things were left unfinished in the generator PR (#593):
- [ ] Movability (for `Unpin` handling)
- [x] Updating the traits table in the book

cc @Aaron1011

### PR DESCRIPTION
**Pull request by Areredify**
pretty straightforward

closes #618
cc #363

### CODE REVIEW

**Review by jackh726 — APPROVED** [review]
I *think* this is correct, but I'm not 100% sure of the rules around this.

Can you update the traits table in the book?

r=me with that

File: book/src/clauses/well_known_traits.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ Some common examples of auto traits are `Send` and `Sync`.
 | slices          | ⚬     | ⚬    | ⚬     |   ✅    | ⚬            | ⚬    | ⚬   | ⚬      |  ⚬       |    ✅       |
 | arrays          | ✅     | ✅    | ✅     |   ❌    | ⚬            | ⚬    | ⚬   | ⚬      |  ⚬       |    ✅        |
 | closures        | ✅     | ✅    | ✅     | ⚬      | ⚬            | ⚬    | ✅   | ⚬      |  ⚬       |    ✅        |
-| generators      |  ⚬    |  ⚬  | ❌     |  ⚬     | ⚬            | ⚬     | ⚬  | ❌      |   ❌       |    ❌       |
+| generators      |  ⚬    |  ⚬  | ❌     |  ⚬     | ⚬            | ⚬     | ⚬  | ✅      |   ❌       |    ❌       |
```
Comment (jackh726) [inline book/src/clauses/well_known_traits.md:47]: Doesn't this also implement auto trait handling for generatos?

Comment (Areredify) [reply book/src/clauses/well_known_traits.md:47]: It was already implemented but I guess it's complete now

**Pull request merged**